src/main/java/com/atlassian/bitbucket/jenkins/internal/model/deployment/DeploymentState.java

```diff
@@ -0,0 +1,17 @@
+package com.atlassian.bitbucket.jenkins.internal.model.deployment;
+
+/**
+ * The possible states of a deployment.
```

Comment:
This feels very spartan to me. There is _nothing_ here to help me decide what things mean. I know writing javadoc is hard but at the very minimum I think we should link perhaps to our public docs (which I know are not public yet).

Reply:
> callers can choose whichever 

I don't think this is true. We don't expose API. It is _here_ (in this plugin) that you assign meaning to these values. From discussions it may even be possible for users to not assign these and the plugin makes the appropriate decisions for itself. This I think it would be good for us to probably document a bit more around what they mean for us here in Jenkins land.